### ISSUE TITLE
Create TodoMVC Example

### ISSUE THREAD

**Issue by fdietze** [OutWatch/outwatch#283]
I just saw this on Reddit and wondered, how elegantly a TodoMVC could be implemented with outwatch.
https://www.reddit.com/r/scala/comments/ai6vcw/todomvc_in_139_lines_of_scala_with_react4s

Does anyone want to give it a go? @cornerman and me are ready to give feedback and make improvements.

**Comment by elyphas** [comment]
In fact, my project is something a todo app, :)

**Comment by cornerman** [comment]
> In fact, my project is something like  a todo app, :)

Nice. Feel free to share it if you feel comfortable with it :+1:

In the best case, we have a very minimal todo app that works similiar to the one in the example on reddit.

**Comment by elyphas** [comment]
Yes of course, the problem i have is i am new in all these technologies,
cats, monix, frp in general but my wish is to translade a project from
scalajs-react to outwatch. Thanks for all the help you had given me

**Comment by fdietze** [comment]
Note: To be included in the official [TodoMVC](http://todomvc.com/) repository, one has to implement the [App Specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md).

**Comment by zakpatterson** [comment]
I got started working on this as a means to learning the library, and ran into a couple issues where I think I'm just not understanding things.

[See source](https://github.com/clovellytech/outwatch-examples/blob/master/todomvc/src/main/scala/todomvc/TodoMvc.scala)

[See issues](https://github.com/clovellytech/outwatch-examples/issues)

[See demo](https://clovellytech.github.io/outwatch-examples/todomvc/)

I think it's pretty close, I'm just not quite getting how some of the handler stuff works.  Please take a look, I'd be very happy to migrate it into your org asap when it's working correctly.

**Comment by zakpatterson** [comment]
This issue for example is confusing me. At the onChange handler, the value of text is still the prior input. So user inputs "123" and hits enter, then a todo with "" is saved. Next save, "123" will store:

```scala
def addTodo(todoHandler: Handler[(Int, List[TodoItem])]): IO[VNode] = Handler.create[String]("").map { text =>
    input(cls := "new-todo",
      placeholder := "What needs to be done?",
      autoFocus,
      value <-- text,
      onChange.target.value --> text,
      text zip todoHandler map { case (title, (nextId, todos)) =>
        onEnterUp.map(_ => (nextId + 1, TodoItem(nextId, title, false) :: todos)) --> todoHandler
      }
    )
  }
```

**Comment by fdietze** [comment]
Yes, Thank you for building examples. I'm also very busy and give you feedback as soon as possible!

**Comment by elyphas** [comment]
I want to publish a little bit of what I am doing but it is not precisely a todo app, but more a list of items to buy, so it is not fixed nicely into this:

https://github.com/tastejs/todomvc/blob/master/app-spec.md

it is worth to do it?, sorry, many things to learn and I having hard time to manage the projects at the same time.

but I feel I am doing much advances, and I am happy to use the awesome outwatch.

**Comment by zakpatterson** [comment]
@elyphas if you have some example of any kind I would welcome it to be included as an example in this project: 

https://github.com/clovellytech/outwatch-examples

I'd be happy to work with you over there to include it.

**Comment by elyphas** [comment]
@zakpatterson ; ah ok, thank you, I am working on it, I hope it will be very soon
